modules/python/src2/cv2_convert.cpp

```diff
@@ -301,6 +301,12 @@ bool pyopencv_to(PyObject* o, Mat& m, const ArgInfo& info)
     m.u = GetNumpyAllocator().allocate(o, ndims, size, type, step);
     m.addref();
 
+    // if the input is a 3d array with shape (n,1,2), we want to
+    // automatically reshape it to a 2d array with shape (n,2) and 1 channel.
+    if (ndims == 3 && _sizes[1] == 1 && _sizes[2] == 2) {
+        m = m.reshape(1, m.size[0]);
+    }
```

Comment:
The change irrelevant.  The documentation says that the function input should be 2d: https://docs.opencv.org/4.x/d9/d0c/group__calib3d.html#gacbba2ee98258ca81d352a31faa15a021. Such hacks in converters destroy shapes of DNN tensors and some other functions related to 3d.